test/tests.js

```diff
@@ -968,16 +968,44 @@ describe( 'passport-saml /', function() {
+        '</samlp:Response>';
+        var base64xml = new Buffer( xml ).toString('base64');
+        var container = { SAMLResponse: base64xml };
+        var samlConfig = {
+          entryPoint: 'https://app.onelogin.com/trust/saml2/http-post/sso/371755',
+          cert: TEST_CERT,
+          validateInResponseTo: true
+        };
+        var samlObj = new SAML( samlConfig );
+
+        // Mock the SAML request being passed through Passport-SAML
+        samlObj.cacheProvider.save(requestId, new Date().toISOString(), function(){});
+
+        samlObj.validatePostResponse( container, function( err, profile, logout ) {
+          should.exist( err );
+          err.message.should.match("Invalid signature");
+          samlObj.validatePostResponse( container, function( err, profile, logout ) {
+            should.exist( err );
```

Comment:
This `done()` is in a callback, so it needs to be in a `try...catch` block, otherwise a failure here will be reported as a timeout when the tests runs.

Reply:
Good catch, fixed!

There was also one Buffer() remaining so I updated it to Buffer.from().